src/main/java/io/dockstore/provision/DOSPluginUtil.java

```diff
@@ -0,0 +1,104 @@
+package io.dockstore.provision;
+
+import com.google.common.collect.Lists;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+
+class DOSPluginUtil {
+
+    private static final String API = "/ga4gh/dos/v1/dataobjects/";
+    private static final int SCHEME = 0;
+    private static final int HOST = 1;
+    private static final int PATH = 2;
+
+    // Package-private constructor
+    DOSPluginUtil() {
+    }
+
+    /**
+     *
+     *
+     * @param dosURI The string targetPath
+     * @return the scheme, host, and path of the targetPath, or <code>Optional.empty()</code>
+     */
+    Optional<ImmutableTriple<String, String, String>> splitUri(String dosURI) {
```

Comment:
I would have used a new class, but there's no point in overdoing this, so maybe in a future PR.

```
static class HostAndPath {
  private final String host;
  private final String targetPath;
  public HostAndPath(String host, String targetPath) {
...
```
That way the compiler would ensure you're using the right field, whereas now you run into the possibility of calling getMiddle() when you meant to call getRight().

But you have tests that should catch any accidental misuses, so I'm good with it.